SingularityService/src/main/java/com/hubspot/singularity/auth/SingularityAuthorizationHelper.java

```diff
@@ -142,7 +142,7 @@ public void checkForAuthorization(SingularityRequest request, Optional<Singulari
 
     final boolean userIsAdmin = adminGroups.isEmpty() ? false : groupsIntersect(userGroups, adminGroups);
     final boolean userIsJITA = jitaGroups.isEmpty() ? false : groupsIntersect(userGroups, jitaGroups);
-    final boolean userIsRequestOwner = request.getGroup().isPresent() ? userGroups.contains(request.getGroup().get()) : true;
+    final boolean userIsRequestOwner = request.getGroup().isPresent() ? groupsIntersect(userGroups, request.getReadWriteGroups().get()) : true;
```

Comment:
we still need to be checking `request.getGroup()` (so essentially, the union of `group` and `readWriteGroups`)